core/vm/interpreter.go

```diff
@@ -234,3 +257,21 @@ func (in *Interpreter) Run(contract *Contract, input []byte) (ret []byte, err er
 	}
 	return nil, nil
 }
+
+// CanRun tells if the contract, passed as an argument, can be
+// run by the current interpreter.
+// The EVM interpreter will try to interpret any binary file and
+// as such, should be tried last.
```

Comment:
Please remove 

```
+// The EVM interpreter will try to interpret any binary file and
+// as such, should be tried last.
```

That will be up to a consensus descision on how to enable it, most probably by having an explicit flag in the accounts specifying whether the code is EVM or EWASM.